cmake/SeracBasics.cmake

```diff
@@ -28,6 +28,9 @@ endif()
 option(SERAC_ENABLE_CODE_CHECKS "Enable Serac's code checks" ${_enable_serac_code_checks})
 option(SERAC_ENABLE_PROFILING "Enable profiling functionality" OFF)
 
+cmake_dependent_option(SERAC_ENABLE_BENCHMARKS "Enable benchmark executables" ON
```

Comment:
Can you add a check here that does something like this:

```
if ((SERAC_ENABLE_BENCHMARKS OR ENABLE_BENCHMARKS) AND NOT SERAC_ENABLE_PROFILING)
    message(FATAL_ERROR "SERAC_ENABLE_BENCHMARKS requires SERAC_ENABLE_PROFILING to be turned on")
endif()
```

So we have a better user experience.